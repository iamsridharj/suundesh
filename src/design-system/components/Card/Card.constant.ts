export const CardTestIds = {
    cardWrapper: "cardWrapper"
}