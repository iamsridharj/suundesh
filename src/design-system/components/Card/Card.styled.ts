import { styled } from "styled-components";

export const StyledCardWrapper = styled.article`
    border-radius: 4px 4px 0 0;
    position: relative;
    height: 100%;
    width: 100%;
    border: 1px solid ${({ theme }) => {
      return theme.colors.raven.raven40
    }};
    background-color: ${({ theme }) => {
      return theme.colors.white
    }};
`