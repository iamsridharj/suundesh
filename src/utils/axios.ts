import axios, { AxiosRequestConfig, AxiosInstance } from 'axios';

const baseURL = "https://dev-api.konfhub.com";

const axiosWrapper: AxiosInstance = axios.create({
    baseURL: baseURL,
    headers: {
        'Content-Type': 'application/json',
    },
});

interface BackendResponse<T> {
  status: string;
  data: T;
}

const get = async <T>(url: string, params: Record<string, any> = {}, config: AxiosRequestConfig = {}): Promise<BackendResponse<T>> => {
  try {
    const response = await axiosWrapper.get<BackendResponse<T>>(url, { params, ...config });
    return response.data;
  } catch (error) {
    throw error;
  }
};

const post = async <T>(url: string, data: any = {}, config: AxiosRequestConfig = {}): Promise<BackendResponse<T>> => {
  try {
    const response = await axiosWrapper.post<BackendResponse<T>>(url, data, { ...config });
    return response.data;
  } catch (error) {
    throw error;
  }
};

export { axiosWrapper, get, post };
