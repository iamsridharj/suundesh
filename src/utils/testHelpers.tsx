// src/utils/testHelpers.tsx
import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { useTheme } from "src/hooks/ui/useTheme";

const RenderWithTheme = ({ children }: { children: React.ReactNode }) => {
  const theme = useTheme();

  return (
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  );
};

const render = (ui: React.ReactElement, options = {}) => {
  return rtlRender(
    <RenderWithTheme>
      {ui}
    </RenderWithTheme>,
    options
  );
};

export * from '@testing-library/react';
export { render };
