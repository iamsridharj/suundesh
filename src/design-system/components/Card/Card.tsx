import React from 'react';
import { StyledCardWrapper } from './Card.styled';
import { CardTestIds } from './Card.constant';

interface CardProps {
    children: React.ReactNode;
}

export const Card: React.FC<CardProps> = ({ children }) => {
    return <StyledCardWrapper data-testid={CardTestIds.cardWrapper} >{children}</StyledCardWrapper>;
};
