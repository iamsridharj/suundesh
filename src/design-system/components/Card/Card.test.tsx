import React from 'react';
import { render } from 'src/utils/testHelpers';
import { Card } from './Card';
import { CardTestIds } from './Card.constant';

describe('Card Component', () => {
    it('renders without crashing', () => {
        const { getByTestId } = render(<Card>Test Content</Card>);
        expect(getByTestId(CardTestIds.cardWrapper)).toBeInTheDocument();
    });

    it('displays the correct children', () => {
        const childText = 'Test Content';
        const { getByText } = render(<Card>{childText}</Card>);
        expect(getByText(childText)).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<Card>Test Content</Card>);
        expect(asFragment()).toMatchSnapshot();
    });
});
