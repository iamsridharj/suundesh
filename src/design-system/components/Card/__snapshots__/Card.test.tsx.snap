// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component matches snapshot 1`] = `
<DocumentFragment>
  <article
    class="sc-beySPh cexEpP"
    data-testid="cardWrapper"
  >
    Test Content
  </article>
</DocumentFragment>
`;
